test/e2e/network/dns.go

```diff
@@ -49,14 +49,22 @@ var _ = SIGDescribe("DNS", func() {
 		namesToResolve := []string{
 			fmt.Sprintf("kubernetes.default.svc.%s", framework.TestContext.ClusterDNSDomain),
 		}
-		// Added due to #8512. This is critical for GCE and GKE deployments.
-		if framework.ProviderIs("gce", "gke") {
-			namesToResolve = append(namesToResolve, "google.com")
-			// Windows containers do not have a route to the GCE
-			// metadata server by default.
-			if !framework.NodeOSDistroIs("windows") {
-				namesToResolve = append(namesToResolve, "metadata")
-			}
+		wheezyProbeCmd, wheezyFileNames := createProbeCommand(namesToResolve, nil, "", "wheezy", f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		jessieProbeCmd, jessieFileNames := createProbeCommand(namesToResolve, nil, "", "jessie", f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		ginkgo.By("Running these commands on wheezy: " + wheezyProbeCmd + "\n")
+		ginkgo.By("Running these commands on jessie: " + jessieProbeCmd + "\n")
+
+		// Run a pod which probes DNS and exposes the results by HTTP.
+		ginkgo.By("creating a pod to probe DNS")
+		pod := createDNSPod(f.Namespace.Name, wheezyProbeCmd, jessieProbeCmd, dnsTestPodHostName, dnsTestServiceName)
+		validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
+	})
+
+	// Added due to #8512. This is critical for GCE and GKE deployments.
+	ginkgo.It("should provide DNS for the cluster [Provider:GCE]", func() {
```

Comment:
@spiffxp  is `[Provider:GCE]` equivalent to `framework.ProviderIs("gce", "gke")` in the e2e framework?

Reply:
I completely made the tag up out of whole cloth but I feel like we're going to want some way to signify that we want these provider-specific tests to run.  I could do something like `[Feature:GCEProvider]` as well, but this would exclude it from all of the release-master-blocking tests, almost all of which run in GCE and need this tested anyway.  Open to feedback on a more appropriate way to do things here

Reply:
I don't see how this can go up for approval right now, this new test will fail on other providers. I think we need to add a skip in the new test if it's not GCE or GKE. I don't know that we need to decide on a label for it right now.

Reply:
This test isn't intended to be a Conformance test though, the Conformance part of it will still pass just fine.  This was in service of allowing other tests to be promoted to Conformance (see https://github.com/kubernetes/kubernetes/pull/74977)

Reply:
But why are we adding a failing e2e test (regardless of any conformance question)?

Reply:
Once we decide on the best way to do this, please update the documentation
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests

Reply:
https://github.com/kubernetes/kubernetes/pull/80682#discussion_r309370687 Addressed.

https://github.com/kubernetes/kubernetes/blob/20010af0de0184410107fb4c7d2f0c4d74db5f15/test/e2e/network/dns.go#L65